src/rest-server/src/templates/dockerContainerScript.mustache

```diff
@@ -177,18 +179,41 @@ function  azure_rdma_preparation()
 azure_rdma_preparation
 {{/ azRDMA }}
 {{/ reqAzRDMA }}
-# Write env to system-wide environment
-env | grep -E "^PAI|PATH|PREFIX|JAVA|HADOOP|NVIDIA|CUDA" > /etc/environment
+
+function write_user_command()
+{
+  local scriptPath="$1"
+
+  # Read an escaped string to an array and unescape
+  local lines=$'{{ taskData.command }}'
+  readarray -n 50 -t lines <<< "${lines}"
+  read -n 1000 -a lines <<< "${lines[@]}"
+
+  echo "#!/bin/bash" > "${scriptPath}"
+
+  # By default, exit on the first error
+  echo 'set -e' >> "${scriptPath}"
+  # Enable logging for the user command
+  echo 'set -x' >> "${scriptPath}"
+
+  printf '%s\n' "${lines[@]}" >> "${scriptPath}"
+
+  chmod a+x "${scriptPath}"
+}
 
 function run_user_command()
 {
+  local scriptPath="$1"
   printf "%s %s\n\n" "[INFO]" "USER COMMAND START"
-  eval {{ taskData.command }} || exit $?
+  exec "${scriptPath}" || exit $?
```

Comment:
exec will overwrite parent context, so `printf "\n%s %s\n\n" "[INFO]" "USER COMMAND END"` will never be executed?

Reply:
You were right, fixed.